include/deal.II/base/function.templates.h

```diff
@@ -106,10 +106,9 @@ Function<dim, RangeNumberType>::vector_value_list(
 {
   // check whether component is in the valid range is up to the derived
   // class
-  Assert(values.size() == points.size(),
-         ExcDimensionMismatch(values.size(), points.size()));
+  // AssertDimension(points.size(), values.size());
 
-  for (unsigned int i = 0; i < points.size(); ++i)
+  for (unsigned int i = 0; i < values.size(); ++i)
```

Comment:
No, this isn't right. If the function is called with a different number of input and output arguments, then the place where the function is called needs to be fixed.